opentracing/utils.go

```diff
@@ -0,0 +1,27 @@
+package opentracing
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+)
+
+const (
+	OpenTracingContextIDHeader = "OpenTracing-Context-Id"
```

Comment:
Yes, totally agree in having util methods for common behavior. Just questioning whether the actual header name needs to be impl specific. I think one of your goals was to keep the Raw API minimal, and have the shared Construction API to do the common tasks. So we want to decide whether serialization of the TraceContext into the on-the-wire protocol is a common task or an implementation detail of the Raw API. If I am not mistaken, current `ContextID` belongs to the Raw API, and it has `toBytes`. If we had the following API in _some_ component of Raw layer, would be be sufficient for serialization?
```Go
type TraceContextPickler interface {
    // takes care of serialization to text headers
    func asStringMap(context TraceContext) map[string]string 

    func fromStringMap(headers map[string]string) (TraceContext, error)

    // takes care of byte stream serialization
    func asBinary(context TraceContext) []byte  // or perhaps accept target byte buffer

    func fromBinary(bytes []byte) (TraceContext, error)
}
```

Then the common util method can be `toRequestHeaders(httpReq, context, pickler)`